DOI.js

```diff
@@ -93,52 +94,83 @@ function completeDOIs(doc) {
 	}
 }
 
-function retrieveDOIs(DOIs, doc) {
-	__num_DOIs = DOIs.length;
 
+var providers = [
+	"11645bd1-0420-45c1-badb-53fb41eeb753", //CrossRef
+	"9f1fb86b-92c8-4db7-b8ee-0b481d456428"  //DataCite
+];
+
+
+/*
+var providers = [
+	"9f1fb86b-92c8-4db7-b8ee-0b481d456428", //DataCite
+	"11645bd1-0420-45c1-badb-53fb41eeb753"  //CrossRef
+];
+*/
+
+function retrieveDOIs(DOIs, doc, provider) {
+	__num_DOIs = DOIs.length;
+	
 	for(var i=0, n=DOIs.length; i<n; i++) {
 		(function(doc, DOI) {
 			var translate = Zotero.loadTranslator("search");
-			translate.setTranslator("11645bd1-0420-45c1-badb-53fb41eeb753");
-	
+			translate.setTranslator(provider);
+			
 			var item = {"itemType":"journalArticle", "DOI":DOI};
 			translate.setSearch(item);
-	
+
 			// don't save when item is done
 			translate.setHandler("itemDone", function(translate, item) {
 				if (!item.title) {
-					Zotero.debug("No title available for " + DOI);
-					return;
+					Zotero.debug("No title available for " + item.DOI);
+					item.title = "[No Title]";
+					//return;
 				}
-				
-				item.repository = "CrossRef";
-				items[DOI] = item;
-				selectArray[DOI] = item.title;
+
+				items[item.DOI] = item;
+				selectArray[item.DOI] = item.title;
+				// done means not remaining anymore
+				Z.debug(item.DOI + "   " + remainingDOIs + "   " + provider + "    " + remainingDOIs.indexOf(item.DOI) );
+				remainingDOIs.splice(remainingDOIs.indexOf(item.DOI), 1);
+				Z.debug(remainingDOIs);
 			});
 	
 			translate.setHandler("done", function(translate) {
 				__num_DOIs--;
 				if(__num_DOIs <= 0) {
-					completeDOIs(doc);
+					Z.debug("done    " + providers.length);
```

Comment:
Here I would display a debug message along the lines of "Done with CrossRef. Remaining DOIs: doi, doi, doi"